text.cabal

```diff
@@ -97,6 +97,15 @@ library
     else
       extra-libraries: stdc++
 
+  -- Certain version of GHC crash on Windows, when TemplateHaskell encounters C++.
+  -- https://gitlab.haskell.org/ghc/ghc/-/issues/19417
+  if flag(simdutf) && os(windows) && (impl(ghc == 8.0.1) || impl(ghc >= 8.8) && impl(ghc < 8.10.5) || impl(ghc == 9.0.1))
+    build-depends: base < 0
+
+  -- For GHC 8.2, 8.6.3 and 8.10.1 even TH + C crash Windows linker.
+  if os(windows) && (impl(ghc >= 8.2) && impl(ghc < 8.4) || impl(ghc == 8.6.3) || impl(ghc == 8.10.1))
```

Comment:
You can write

```cabal
  -- For GHC 8.2, 8.6.3 and 8.10.1 even TH + C crash Windows linker.
  if os(windows) && impl(ghc (>= 8.2 && < 8.4) || ==8.6.3 || == 8.10.1)
    build-depends: base < 0
```

i.e. use same version range syntax as in `build-depends`. That is more readable IMO.

Reply:
Ah, thanks, great!